source/npm/qsharp/ux/circuit-vis/formatters/formatUtils.ts

```diff
@@ -98,12 +98,21 @@ export const circle = (
  *
  * @param x      x coord of circle.
  * @param y      y coord of circle.
- * @param radius Radius of circle.
+ * @param wireYs y coords of wire(s) connected to the control dot.
  *
  * @returns SVG element for control dot.
  */
-export const controlDot = (x: number, y: number, radius = 5): SVGElement =>
-  circle(x, y, radius, "control-dot");
+export const controlDot = (
+  x: number,
+  y: number,
+  wireYs: number[],
+): SVGElement => {
+  const radius = 5;
+  const dot = circle(x, y, radius, "control-dot");
+  dot.setAttribute("data-wire-ys", JSON.stringify(wireYs));
+  dot.setAttribute("data-width", `${2 * radius}`);
```

Comment:
This is an example of where we set a `data-` attribute _before_ rendering the circuit, instead of using `getBBox` _after_ rendering to compute this information.